wgpu-hal/src/dx12/adapter.rs

```diff
@@ -76,45 +79,41 @@ impl super::Adapter {
 
         // Detect the highest supported feature level.
         let d3d_feature_level = [
-            d3d12::FeatureLevel::L12_1,
-            d3d12::FeatureLevel::L12_0,
-            d3d12::FeatureLevel::L11_1,
-            d3d12::FeatureLevel::L11_0,
+            Direct3D::D3D_FEATURE_LEVEL_12_1,
+            Direct3D::D3D_FEATURE_LEVEL_12_0,
+            Direct3D::D3D_FEATURE_LEVEL_11_1,
+            Direct3D::D3D_FEATURE_LEVEL_11_0,
         ];
-        let mut device_levels: d3d12_ty::D3D12_FEATURE_DATA_FEATURE_LEVELS =
+        let mut device_levels: Direct3D12::D3D12_FEATURE_DATA_FEATURE_LEVELS =
             unsafe { mem::zeroed() };
         device_levels.NumFeatureLevels = d3d_feature_level.len() as u32;
         device_levels.pFeatureLevelsRequested = d3d_feature_level.as_ptr().cast();
         unsafe {
             device.CheckFeatureSupport(
-                d3d12_ty::D3D12_FEATURE_FEATURE_LEVELS,
-                ptr::from_mut(&mut device_levels).cast(),
-                mem::size_of::<d3d12_ty::D3D12_FEATURE_DATA_FEATURE_LEVELS>() as _,
+                Direct3D12::D3D12_FEATURE_FEATURE_LEVELS,
+                <*mut _>::cast(&mut device_levels),
+                mem::size_of_val(&device_levels) as u32,
             )
-        };
-        // This cast should never fail because we only requested feature levels that are already in the enum.
-        let max_feature_level =
-            d3d12::FeatureLevel::try_from(device_levels.MaxSupportedFeatureLevel)
-                .expect("Unexpected feature level");
+        }
+        .unwrap();
+        let max_feature_level = device_levels.MaxSupportedFeatureLevel;
 
         // We have found a possible adapter.
         // Acquire the device information.
-        let mut desc: dxgi1_2::DXGI_ADAPTER_DESC2 = unsafe { mem::zeroed() };
-        unsafe {
-            adapter.unwrap_adapter2().GetDesc2(&mut desc);
-        }
+        let desc = unsafe { adapter.unwrap_adapter2().GetDesc2() }.unwrap();
 
         let device_name = auxil::dxgi::conv::map_adapter_name(desc.Description);
 
-        let mut features_architecture: d3d12_ty::D3D12_FEATURE_DATA_ARCHITECTURE =
+        let mut features_architecture: Direct3D12::D3D12_FEATURE_DATA_ARCHITECTURE =
             unsafe { mem::zeroed() };
```

Comment:
Can't this be default() instead of zeroed?

Reply:
Thanks for the reminder, replaced all `unsafe { mem::zeroed() }` where sensible (and helping the compiler to catch one more trivial FRU initializer).